staker/block_validator_instance.go

```diff
@@ -0,0 +1,472 @@
+package staker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rlp"
+
+	"github.com/offchainlabs/nitro/arbutil"
+	"github.com/offchainlabs/nitro/execution"
+	"github.com/offchainlabs/nitro/validator"
+	"github.com/offchainlabs/nitro/validator/client/redis"
+	"github.com/offchainlabs/nitro/validator/inputs"
+)
+
+var _ ValidatorInstance = (*BlockValidatorInstance)(nil)
+var _ LegacyValidatorInstance = (*BlockValidatorInstance)(nil)
+
+type BlockValidatorInstance struct {
```

Comment:
We should probably have doc comments on every struct in our codebase. Especially ones that are publicly exported by the package.